main.cpp

```diff
@@ -52,3 +58,29 @@ int main(int argc, char *argv[])
     return SailfishApp::main(argc, argv);
 }
 
+#else
+// Example 2: use custom QQuickView
+int main(int argc, char *argv[])
+{
+    // FIXME: keeps crashing! -> provide a working example
```

Comment:
The problem here is that Sailfish uses a booster (mapplauncherd). It preloads a process that instantiates QGuiApplication and QQuickView (or depends a bit on booster type) and when application gets actually launched, those preinstantiated things are set to load qml source. Desktop file has "X-Nemo-Application-Type=silica-qt5" which makes such trigger.

So what the app should do is export the main function and use mdeclarativecache like e.g. in https://github.com/sailfishos/sailfish-browser/blob/master/src/main.cpp

Reply:
Thanks for this info! I will split the topic into another "future" branch then.